airflow/utils/helpers.py

```diff
@@ -202,3 +203,9 @@ def cross_downstream(*args, **kwargs):
         stacklevel=2,
     )
     return import_string('airflow.models.baseoperator.cross_downstream')(*args, **kwargs)
+
+
+def is_k8s_or_k8scelery_executor() -> bool:
+    """Determines if the executor utilizes k8s."""
+    conf_executor = conf.get('core', 'EXECUTOR')
+    return conf_executor in {'KubernetesExecutor', 'CeleryKubernetesExecutor'}
```

Comment:
This was not addressed. We should use `ExecutorLoader.KubernetesExecutor` instead of hardcoding the name. What is more I would be in favour of two things:
a) move this to `executor_loader.py` module
b) Create constant KUBERNETES_EXECUTORS = {...}

Then we can do
```py
def is_k8s_or_k8scelery_executor() -> bool:
   return conf.get('core', 'EXECUTOR') in KUBERNETES_EXECUTORS
```

No strong opinion if this should be a separate function or `ExecutorsLoader` method.

Reply:
@turbaszek any idea how we can do that without creating a circular import? that's what happened when we tried.

Reply:
Can you tlell what were the circular imports. Usually (not always) circular imports can be resolved via applying single responsibility principle and splitting out or joining certain classes. 

Usually, I solve them in the way that I put closely coupled classes together - in a single module, but those classes that are not coupled in separate modules and make sure one those modules do not import each other. Sometimes (and usually it results in a cleaner architecture and better "single responsibility principle" application) it results in having to add some kind of mediator between those classes. 

However in this case we have a big architectural problem as the "conf" module (which should be merely an util) pulls in half of the airlfow (because it is not designed well to be "just conf"). I tried several times to untangle it with varying level of success, and I am afraid it might be difficult to solve. It's rather difficult and result of historical way of implementing the conf class. I think we should finally solve it at some point in time, but this is quite an effort as it ripples through half of the code of airflow (which is bad IMHO but we have to leave with it until we fix it).

Reply:
@potiuk @turbaszek would it be sufficient if I just make ExecutorLoader an import within the function? Though I'm not sure if that would cause runtime issues...

Reply:
So the problem is tied to the fact that ExecutorLoader imports BaseExecutor, which imports TaskInstance.... which imports helpers... 

I implemented one potential solution which is to move these constants out of the ExecutorLoader (which kinda makes sense since we're not loading an executor).

Reply:
Yeah. This chain of dependencies is exactly the problem. Constants out of the loader is a good direction 👍

Reply:
Also imports inside a function (almost) never create an import cycle.

Reply:
> Also imports inside a function (almost) never create an import cycle.

But I am of the opinion that often it is hiding the problem rather than solving it. But that's my opinion, not a fact.